services/shhext/publisher/service.go

```diff
@@ -0,0 +1,446 @@
+		return nil, err
+	}
+
+	whisperMessage, err := s.directMessageToWhisper(privateKey, publicKey, msg.PubKey, msg.Sig, msgSpec)
+	if err != nil {
+		s.log.Error("sshext-service", "error building whisper message", err)
+		return nil, err
+	}
+
+	return s.whisperAPI.Post(ctx, *whisperMessage)
+}
+
+func (s *Service) directMessageToWhisper(myPrivateKey *ecdsa.PrivateKey, theirPublicKey *ecdsa.PublicKey, destination hexutil.Bytes, signature string, spec *chat.ProtocolMessageSpec) (*whisper.NewMessage, error) {
+	// marshal for sending to wire
+	marshaledMessage, err := proto.Marshal(spec.Message)
+	if err != nil {
+		s.log.Error("encryption-service", "error marshaling message", err)
+		return nil, err
+	}
+
+	whisperMessage := whisperutils.DefaultWhisperMessage()
+	whisperMessage.Payload = marshaledMessage
+	whisperMessage.Sig = signature
+
+	if spec.SharedSecret != nil {
+		chat := s.GetNegotiatedChat(theirPublicKey)
+		if chat != nil {
+			s.log.Debug("Sending on negotiated topic")
+			whisperMessage.SymKeyID = chat.SymKeyID
+			whisperMessage.Topic = chat.Topic
+			whisperMessage.PublicKey = nil
+			return &whisperMessage, nil
+		}
+	} else if spec.PartitionedTopic() {
+		s.log.Debug("Sending on partitioned topic")
+		// Create filter on demand
+		if _, err := s.filter.LoadPartitioned(myPrivateKey, theirPublicKey, false); err != nil {
+			return nil, err
+		}
+		t := filter.PublicKeyToPartitionedTopicBytes(theirPublicKey)
+		whisperMessage.Topic = whisper.BytesToTopic(t)
+		whisperMessage.PublicKey = destination
+		return &whisperMessage, nil
+	}
+
+	s.log.Debug("Sending on old discovery topic")
+	whisperMessage.Topic = whisperutils.DiscoveryTopicBytes
+	whisperMessage.PublicKey = destination
+
+	return &whisperMessage, nil
+}
+
+// SendPublicMessage sends a public chat message to the underlying transport
+func (s *Service) SendPublicMessage(ctx context.Context, msg chat.SendPublicMessageRPC) (hexutil.Bytes, error) {
+	if !s.config.PfsEnabled {
+		return nil, ErrPFSNotEnabled
+	}
+
+	filter := s.filter.GetByID(msg.Chat)
+	if filter == nil {
+		return nil, errors.New("not subscribed to chat")
+	}
+
+	// Enrich with transport layer info
+	whisperMessage := whisperutils.DefaultWhisperMessage()
+	whisperMessage.Payload = msg.Payload
+	whisperMessage.Sig = msg.Sig
+	whisperMessage.Topic = whisperutils.ToTopic(msg.Chat)
+	whisperMessage.SymKeyID = filter.SymKeyID
+
+	// And dispatch
+	return s.whisperAPI.Post(ctx, whisperMessage)
+}
+
+func (s *Service) ConfirmMessagesProcessed(ids [][]byte) error {
+	return s.protocol.ConfirmMessagesProcessed(ids)
+}
+
+func (s *Service) startTicker() {
+	s.ticker = time.NewTicker(tickerInterval * time.Second)
+	s.quit = make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-s.ticker.C:
+				err := s.perform()
+				if err != nil {
+					s.log.Error("could not execute tick", "err", err)
+				}
+			case <-s.quit:
+				s.ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (s *Service) perform() error {
```

Comment:
hmm, why is this method doing nothing?

Reply:
I am moving still some parts from status-react to status-go, filling it in as of now